src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -91,95 +91,42 @@ class PSyDataNode(Statement):
     # PSyData interface Fortran module
     fortran_module = "psy_data_mod"
     # The symbols we import from the PSyData Fortran module
-    symbols = ["PSyDataType"]
+    _PSyDataSymbol = namedtuple("_PSyDataSymbol", "name symbol_type")
+    symbols = [_PSyDataSymbol("PSyDataType", TypeSymbol)]
```

Comment:
Do we need this to be class attributes instead of instance attributes? For instance if we have 2 profiling regions with 2 different ProfileNode classes?

Also why do we need the nametuple, can we just have: `self._symbols = [TypeSymbol("PSyDataType")]`
and use `self._symbols[0].name` in the gen_code()?

Finally, we only ever use this symbol, is this going to change in the future?, otherwise it could be:
`self._typesymbol = TypeSymbol("PSyDataType")`
or even better `self._typesymbol = TypeSymbol(self._text_name + "Type")` but this will require changes in the tests.

To be honest I can't see clearly where this is needed, so I don't know exactly what is the best option.

Reply:
I've re-worked this a fair bit. I think you were right about needing instance attributes so I've added those and stored the full names (ie with prefixes added) in them. I've also added a test that we can have two different prefixes in the same invoke.
We can't currently work with `Symbols` because this construction of the names happens in the PSyDataNode constructor which doesn't have any access to a symbol table. (This means we can't solve any name clashes for symbols imported from the PSyData Fortran module (which we could do in principle). If there's a clash with the name of the PSyData Fortran module itself then I don't think there's anything we can do anyway.) I went for the namedtuple because I always think that trying to understand code that just does e.g. `symbol[0]` is harder than `symbol.name`. It is slightly overkill in this situation but I don't think it hurts.

Reply:
> We can't currently work with Symbols because this construction of the names happens in the PSyDataNode constructor which doesn't have any access to a symbol table.

I see, that makes things complicated but I think its fine for now, al least while we have the gen_code and update.

>  I went for the namedtuple because I always think that trying to understand code that just does e.g. symbol[0] is harder than symbol.name. It is slightly overkill in this situation but I don't think it hurts.

I think that is a good reason to use namedtuple, makes for better code.

One issue remaining is that we don't use its contents as much as possible. Just in the same file there is the "PSyDataType" hardcoded in 4 different places, which would make changing the attribute break things I believe. Why is this a list? Do we expect more things to be in this list? And why is _psydata_fortran_module a special case and not in the list?

Alternatively, since the type is also mandatory for all PSyData classes, why don't we put the type in a _psydata_fortran_type equivalent to the module so then when we need its string it can be unequivocally referenced? Then we can get rid of the _psydata_symbols unless there is a use case for it. Also then we don't need to say is a TypeSymbol here (as we don't say the fortran_module is a ContainerSymbol) as this is known at the point of use.

Also depending on my other comment on the prefix. If this turns out to be static why not also construct these names statically, e.g. _psydata_fortran_module = _prefix + "psy_data_mod"